shared/elements/gaia_lowstorage_dialog/script.js

```diff
@@ -0,0 +1,78 @@
+'use strict';
+/* global ComponentUtils, MozActivity */
+
+/**
+ * The gaia-confirm component displays a dialog in which the user has a
+ * choice to confirm or cancel the action. It may be displayed along with a
+ * title, description, and image. Buttons may also be configured.
+ * @requires GaiaButtons
+ */
+window.GaiaLowstorageDialog = (function(win) {
+  // Extend from the HTMLElement prototype
+  var proto = Object.create(HTMLElement.prototype);
+
+  // Allow baseurl to be overridden (used for demo page)
+  var baseurl = window.GaiaLowstorageDialogBaseurl ||
+    '/shared/elements/gaia_lowstorage_dialog/';
+
+  function learnmore() {
+    var activity = new MozActivity({
+      name: 'configure',
+      data: {
+        target: 'storage',
+        section: 'application storage'
+      }
+    });
+
+    return Promise.resolve(activity);
+  }
+
+  /**
+   * Localize the component manually as l10n attributes are not supported
+   * within the shadow dom. See also: bug 1026236.
+   */
+  proto._localizeShadowDom = function() {
+    navigator.mozL10n.translateFragment(this.shadowRoot);
+  };
+
+  proto.createdCallback = function() {
+    var shadow = this.createShadowRoot();
+    this._template = template.content.cloneNode(true);
+
+    shadow.appendChild(this._template);
```

Comment:
yeah I don't know yet either ;) 

The new-style components are doing: https://github.com/gaia-components/gaia-component/blob/master/gaia-component.js#L131
I'll try this instead